BridgeCareApp/BridgeCare/DataAccessLayer/InvestmentStrategies.cs

```diff
@@ -18,25 +21,96 @@ public InvestmentStrategies(BridgeCareContext context)
 
         public IQueryable<InvestmentStrategyModel> GetInvestmentStrategies(NetworkModel network, BridgeCareContext db)
         {
-            var investmentStrategies = Enumerable.Empty<InvestmentStrategyModel>();
             try
             {
-                investmentStrategies = db.INVESTMENTs
-                    .Select(p => new InvestmentStrategyModel
+                var simulation = db.SIMULATIONS
+                    .Include(d => d.INVESTMENTS)
+                    .Include(d => d.YEARLYINVESTMENTs)
+                    .Select(p => new InvestmentStrategyModel()
                     {
-                        Id = p.SIMULATIONID,
-                        BudgetOrder = p.BUDGETORDER,
-                        FirstYear = p.FIRSTYEAR ?? 0,
-                        NumberYears = p.NUMBERYEARS ?? 0,
-                        InflationRate = p.INFLATIONRATE ?? 0,
-                        DiscountRate = p.DISCOUNTRATE ?? 0
-                    });
+                        Name = p.SIMULATION1,
+                        Description = p.COMMENTS,
+                        SimulationId = p.SIMULATIONID,
+                        NetworkId = p.NETWORKID ?? 0,
+                        FirstYear = p.INVESTMENTS.FIRSTYEAR ?? 0,
+                        NumberYears = p.INVESTMENTS.NUMBERYEARS ?? 0,
+                        InflationRate = p.INVESTMENTS.INFLATIONRATE ?? 0,
+                        DiscountRate = p.INVESTMENTS.DISCOUNTRATE ?? 0,
+                        BudgetOrder = p.INVESTMENTS.BUDGETORDER,
+                        YearlyBudgets = p.YEARLYINVESTMENTs.Select(m => new InvestmentStrategyYearlyBudgetModel
+                        {
+                            Year = m.YEAR_,
+                            Budget = p.YEARLYINVESTMENTs
+                            .Where(n => n.YEAR_ == m.YEAR_)
+                            .Select(f => new InvestmentStrategyBudgetModel()
+                            {
+                                Amount = f.AMOUNT,
+                                Name = f.BUDGETNAME
+                            }).ToList()
+                        }).ToList()
+                    }).ToList();
+
+                //Cannot do this inside the linq query, would be nice
+                foreach (InvestmentStrategyModel model in simulation)
+                {
+                    model.SetBudgets();
+                }
+
+                return simulation.AsQueryable();
+            }
+            catch (SqlException ex)
+            {
+                HandleException.SqlError(ex, "Investment Strategies");
+            }
+            return Enumerable.Empty<InvestmentStrategyModel>().AsQueryable();
+        }
+
+        public bool SetInvestmentStrategies(InvestmentStrategyModel data, BridgeCareContext db)
```

Comment:
Why does this return a bool? Does it succeed in setting an investment strategy?

Reply:
That is the intention.

Technically the way the code is written NOT throwing an exception gets to the return true which indicates it succeeded in writing the investment strategy for every case I can think of.